### PR TITLE
[TASK] Contacts: Add name/address sort & contact search (RT-1752)

### PR DESCRIPTION
**Pull request by rht**
out

### CODE REVIEW

**Comment by mrajvanshy** [comment]
LGTM

**Comment by rht** [comment]
2015-01-06-102802_1162x80_scrot

**Comment by rippletravis** [comment]
Looks great.

File: src/js/directives/directives.js [hunk 1]
```diff
@@ -822,46 +822,54 @@ module.directive('rpSpanSpacing', [function () {
...
-        var sortFieldGetter = $parse(attr.rpOrdersSortHeader);
-        var sortReverse = $parse(attr.rpOrdersSortHeaderReverse);
-        var fieldName = attr.rpOrdersSortHeaderField;
+        var sortFieldGetter = $parse(attr.rpSortHeader);
+        var sortReverse = $parse(attr.rpSortHeaderReverse);
+        var fieldName = attr.rpSortHeaderField;
+
+        function setArrowClass(sorted, isUp) {
+          var i = element.find('i');
+          i[sorted ? 'addClass' : 'removeClass']('sorted');
```
Comment (jublo) [inline src/js/directives/directives.js]: Smart coding. Have you thought about `toggleClass`?

File: src/js/filters/filters.js [hunk 2]
```diff
@@ -523,3 +523,29 @@ module.filter('rpsortmyorders', function () {
     return arrayCopy;
   }
 });
+
+/**
+ * Contacts sorting filter
+ */
+module.filter('rpsortcontacts', function () {
```
Comment (jublo) [inline src/js/filters/filters.js]: No space needed after `function`.

File: src/js/directives/directives.js (hunk 1 above)
Comment (rht) [inline src/js/directives/directives.js]: It reduces everything to 2 lines with ```toggleClass```!

File: src/js/directives/directives.js [hunk 3]
```diff
@@ -871,46 +871,48 @@ module.directive('rpSpanSpacing', [function () {
...
-        var sortReverse = $parse(attr.rpOrdersSortHeaderReverse);
-        var fieldName = attr.rpOrdersSortHeaderField;
+        var sortFieldGetter = $parse(attr.rpSortHeader);
+        var sortReverse = $parse(attr.rpSortHeaderReverse);
+        var fieldName = attr.rpSortHeaderField;
+
+        function setArrowClass(sorted, isUp) {
+          var i = element.find('i');
+          i.toggleClass('sorted', sorted);
+          i.toggleClass(isUp ? 'fa-caret-up' : 'fa-caret-down');
```
Comment (darkdarkdragon) [inline src/js/directives/directives.js]: @rht this will end up with div having both 'fa-caret-up' and 'fa-caret-down' set

Comment (rht) [inline src/js/directives/directives.js]: I have split into 2 for ```toggleClass``` for ```fa-caret-up``` and ```fa-caret-down```

**Comment by vhpoet** [comment]
LGTM

**Pull request merged**